jsk_robot_common/jsk_robot_startup/test/test-smach-utils.l

```diff
@@ -0,0 +1,29 @@
+(require :unittest "lib/llib/unittest.l")
+(require :smach-utils "package://jsk_robot_startup/euslisp/smach-utils.l")
+
+
+(defmacro def-state-func-test (func-name result)
+  `(deftest ,(intern (format nil "~A-test" func-name)) ()
+     (let* ((func '(,func-name '((description . "hoge") (image . "fuga"))))
+            (output (eval func)))
+       (assert (equal output ,result)
+               (format nil "~%func: ~A error, output: ~A~%" ,(format nil "~A" func-name) output)))))
```

Comment:
Another example with more checks.

```lisp
(defmacro def-state-func-test (func-name)
  `(deftest ,(intern (string-upcase (format nil "~A-test" func-name))) ()
     ,@(mapcar #'(lambda (out expected)
                   `(let ((output (,func-name ',out)))
                      (assert (equal output ',expected)
                              (format nil ,(format nil "~~%func: ~A error, output: ~~S~~%" func-name) output))))
               (list '((description . "hoge") (image . "fuga"))
                     '((description . "hoge") (image . "fuga") (other . "value"))
                     '((description . "hoge"))
                     nil)
               (list '((description . "") (image . ""))
                     '((description . "") (image . "") (other . "value"))
                     '((description . ""))
                     nil))))
```

- Prefer upcase strings with intern
- No need to use eval, we can just paste the code in
- Prefer ~S in assert formatting
```
~A: func: state-func error, output: ((description . ) (image . ))
~S: func: state-func error, output: ((description . "") (image . ""))
```